bundler/lib/bundler/ui/shell.rb

```diff
@@ -76,15 +77,23 @@ def quiet?
         level("quiet")
       end
 
+      def interactive?
+        @interactive
+      end
+
       def ask(msg)
         @shell.ask(msg)
       end
 
       def yes?(msg)
+        raise "Cannot ask yes? with a non-interactive shell" unless interactive?
+
         @shell.yes?(msg)
       end
 
-      def no?
+      def no?(msg)
```

Comment:
Unrelated to this PR, I discovered that `no?` was missing a `msg` parameter and added it.

Reply:
Wow, so I guess nobody has ever used this method 😅. Thanks for fixing it!